### PR TITLE
refactor: Prevent casting LSP17 Data shorter than 20 bytes

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?
## ♻️ Refactor

Prevent casting LSP17 Data shorter than 20 bytes.
In this way, if some mistake happens while setting the extension, the call won't be forwarded to the faulty casted extension.

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

File: tests/LSP17ContractExtension/LSP17Extendable.behaviour.ts [hunk 1]
```diff
@@ -911,6 +911,35 @@ export const shouldBehaveLikeLSP17 = (buildContext: () => Promise<LSP17TestConte
...
 
+    describe('edge cases', () => {
+      describe('when setting less than 20 bytes as data value for the LSP17Extension data key', () => {
+        const randomSelector = ethers.utils.hexlify(ethers.utils.randomBytes(4));
+        const randomBytes10Value = ethers.utils.hexlify(ethers.utils.randomBytes(10));
+
+        const lsp17DataKey =
+          ERC725YDataKeys.LSP17.LSP17ExtensionPrefix +
+          randomSelector.substring(2) +
+          '00000000000000000000000000000000';
```
Comment (CJ42) [inline tests/LSP17ContractExtension/LSP17Extendable.behaviour.ts:922]: Suggested change as as pointed out by @b00ste , to avoid literal value errors + for better readability

```suggestion
          '00'.repeat(16);
```

**Review by CJ42 — APPROVED** [review]

**Review by skimaharvey — APPROVED** [review]

**Review by b00ste — APPROVED** [review]
